### PR TITLE
retrieve pipeline requirement within running step

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes
This code is to get running pipeline requirements with a step environment. 

This will be mainly used for the custom code deployment and particularly building docker image for custom deployment when a local orchestrator is used! Unlike the remote orchestrator where we can use the same created docker image to run the pipeline on the deployment, the local orchestrator requires building a new docker image and this build happens when the custom deployment step is running, for that we need the requirement to be already within step environment.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/advanced-guide/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/steps/utils.py
Comment (schustmi) [inline src/zenml/steps/utils.py:457]: Can we refactor this in some function which can be used both here and in the step executor operator (which has code very similar to this)?

Comment (safoinme) [reply src/zenml/steps/utils.py:457]: @schustmi  I think it's possible but somehow I would want to leave it like this for the moment as refactor this, would change some stuff related to the step that would require some more testing :)

File: tests/unit/test_environment.py
Comment (fa9r) [inline tests/unit/test_environment.py:52]: Why is this needed? Shouldn't line 710 of steps/utils.py already take care of this?

**Review by fa9r — APPROVED** [review]
The code itself looks good to me. 

One question I have: when creating a pipeline you can currently specify `requirements=...` and `required_integrations=...` separately. Are both of those already included in the output of `collect_requirements()`? maybe @schustmi knows more about that?

Also, are these changes still valid after #794 gets merged?

TL;DR: code looks good, but should wait for @schustmi 's review before merging :)

File: tests/unit/test_environment.py
Comment (safoinme) [reply tests/unit/test_environment.py]: I don't think it does because that's just for testing of StepEnvironment, it tries to create a new object from that class. also, that list could be just empty rather than have "zenml" on it!

**Comment by schustmi** [comment]
@fa9r @safoinme My PR will indeed change the way those requirements are configured and stored, but we can go ahead and merge this and I'll take care of the rest.

File: tests/unit/test_environment.py
Comment (schustmi) [reply tests/unit/test_environment.py]: I think an empty list would be nicer indeed just to show that this is a "dummy" value.

**Review by schustmi — APPROVED** [review]

**Pull request merged**
